auth_brute_force/controllers/controllers.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 GRAP - Sylvain LE GAL
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import fields, http, registry, SUPERUSER_ID
+from odoo.api import Environment
+from odoo.http import request
+from odoo.addons.web.controllers.main import Home, ensure_db
+
+_logger = logging.getLogger(__name__)
+
+
+class LoginController(Home):
+    @http.route()
+    def web_login(self, redirect=None, **kw):
+        if request.httprequest.method == 'POST':
+            ensure_db()
+            remote = request.httprequest.remote_addr
+            # Get registry and cursor
+            with registry(request.session.db).cursor() as cursor:
+                env = Environment(cursor, SUPERUSER_ID, {})
+                config_obj = env['ir.config_parameter']
+                attempt_obj = env['res.authentication.attempt']
+                banned_remote_obj = env['res.banned.remote']
+                # Get Settings
+                max_attempts_qty = int(config_obj.search_read(
+                    [('key', '=', 'auth_brute_force.max_attempt_qty')],
+                    ['value'])[0]['value'])
+                # Test if remote user is banned
+                banned = banned_remote_obj.search([('remote', '=', remote)])
+                if banned:
+                    request.params['password'] = ''
+                    _logger.warning(
+                        "Authentication tried from remote '%s'. The request "
+                        "has been ignored because the remote has been banned "
+                        "after %d attempts without success. Login tried : '%s'"
+                        "." % (remote, max_attempts_qty,
+                               request.params['login']))
+                else:
+                    # Try to authenticate
+                    result = request.session.authenticate(
+                        request.session.db, request.params['login'],
+                        request.params['password'])
+                # Log attempt
+                cursor.commit()
```

Comment:
There is a commit just after the next instruction, is this commit really needed ?